contract-variable-monitor/src/agent.spec.js

```diff
@@ -0,0 +1,431 @@
+const mockContract = {};
+
+// combine the mocked provider and contracts into the ethers import mock
+jest.mock('forta-agent', () => ({
+  ...jest.requireActual('forta-agent'),
+  getEthersProvider: jest.fn(),
+  ethers: {
+    ...jest.requireActual('ethers'),
+    providers: {
+      JsonRpcBatchProvider: jest.fn(),
+    },
+    Contract: jest.fn().mockReturnValue(mockContract),
+  },
+}));
+
+const {
+  Finding, FindingType, FindingSeverity, ethers,
+} = require('forta-agent');
+
+const { provideHandleBlock, provideInitialize } = require('./agent');
+const {
+  getObjectsFromAbi, getFunctionFromConfig, getRandomCharacterString,
+} = require('./test-utils');
+const utils = require('./utils');
+const config = require('../agent-config.json');
+
+const checkThresholdSpy = jest.spyOn(utils, 'checkThreshold');
+
+// check the configuration file to verify the values
+describe('check agent configuration file', () => {
+  it('procotolName key required', () => {
+    const { protocolName } = config;
+    expect(typeof (protocolName)).toBe('string');
+    expect(protocolName).not.toBe('');
+  });
+
+  it('protocolAbbreviation key required', () => {
+    const { protocolAbbreviation } = config;
+    expect(typeof (protocolAbbreviation)).toBe('string');
+    expect(protocolAbbreviation).not.toBe('');
+  });
+
+  it('developerAbbreviation key required', () => {
+    const { developerAbbreviation } = config;
+    expect(typeof (developerAbbreviation)).toBe('string');
+    expect(developerAbbreviation).not.toBe('');
+  });
+
+  it('contracts key required', () => {
+    const { contracts } = config;
+    expect(typeof (contracts)).toBe('object');
+    expect(contracts).not.toBe({});
+  });
+
+  it('contracts key values must be valid', () => {
+    const { contracts } = config;
+    Object.keys(contracts).forEach((key) => {
+      const { address, abiFile, variables } = contracts[key];
+
+      // check that the address is a valid address
+      expect(ethers.utils.isHexString(address, 20)).toBe(true);
+
+      // load the ABI from the specified file
+      // the call to getAbi will fail if the file does not exist
+      const abi = utils.getAbi(abiFile);
+
+      // get all of the function objects from the loaded ABI file
+      const functionObjects = getObjectsFromAbi(abi, 'function');
+
+      // for all of the variable names specified, verify that their corresponding getter function
+      // exists in the ABI
+      Object.keys(variables).forEach((variableName) => {
+        expect(Object.keys(functionObjects).indexOf(variableName)).not.toBe(-1);
```

Comment:
Do we want to check that the output variable type is numeric?

Reply:
Good point